### ISSUE TITLE
Trackers

### ISSUE THREAD

**Issue by Donkey-Doug** [michaldrabik/showly#537]
The app containsthe following trackers

- Google firebase analytics
- Google crashlytics

Is it maybe possible to remove these?

**Comment by michaldrabik** [comment]
@Donkey-Doug Not planning on removing those as they are 2 main tools that help with tracking bugs and gather basic statistics which help me fix and improve the app. These SDKs are well known and used in basically every app so nothing to be afraid of.

**Comment by michaldrabik** [comment]
Never heard of it but might be worth looking into in the future. At the moment if you wish I can prepare an APK for you with all the Google stuff removed.

**Comment by IzzySoft** [comment]
So any chance you'll implement your promise, @michaldrabik? Else I'll have to close the request for inclusion with my repo, as the number of offending libraries exceed the maximum tolerated.

As you wrote, those SDKs are well known – especially by privacy researchers and proponents where they always raise at least one eyebrow if not both, and so there's indeed reason to be afraid of them especially because they are present in so many apps (about every 2nd app uses at least one of them, so a lot of data from different apps accumulate at a single place and make the users quite transparent in terms of what apps they use, how they use them, when, where etc.pp.)

To put it a bit into the extreme: replace "SDK" by "weapon" and "app" by "soldier", then read your sentence again – and you might feel what we privacy-folk feel. Just being wide-spread doesn't make a thing safe or less worrysome :wink:

**Comment by IzzySoft** [comment]
As this is open for 8 months now, and still unclear if it will be addressed (the currently latest release still has the trackers, and my question on this was still not answered), I'll now have to close the inclusion request with a "decline". Should this one day be fixed, please give me a ping and I can check again. Thanks!

**Comment by BradNut** [comment]
@michaldrabik Any update on removing the trackers from the app so it can be included other repos like F-Droid or @IzzySoft?

Just wondering given no update and the README still claims "soon also on F-Droid".

**Comment by 1RandomDev** [comment]
Since there doesn't seem any activity I created a FOSS fork of the app myself. Because this version uses a lot of Google Services it's probably really difficult to make one project that can be built with and without proprietary software. If someone is interested, the project can be found here https://github.com/1RandomDev/showly-oss

**Comment by 1RandomDev** [comment]
Nice, maybe you should add to the description that Push Notifications and Billing is not working since I removed all Firebase and Google Play libraries.

**Comment by IzzySoft** [comment]
I don't think it needs mentioning that there's no billing :wink: As for notifications: what are those for in the non-free build? Maybe @michaldrabik could be convinced to switch to [UnifiedPush](https://unifiedpush.org/) (and have the FirebaseDistributor integrated with the PlayStore build so folks there wouldn't even notice the difference)?

**Comment by 1RandomDev** [comment]
I have no clue what the push notifications do. I never got one on my phone. Looks like it's just subscribing to the `general` and `shows` channel and showing everything that gets sent to them https://github.com/michaldrabik/showly-2.0/blob/5beb8db4570abf9a872e685425e6571c263bcd0e/app/src/main/java/com/michaldrabik/showly2/ui/main/cases/MainInitialsCase.kt#L55

**Comment by IzzySoft** [comment]
Well, if you never got one, nothing is missing now, right? :see_no_evil:

**Comment by michaldrabik** [comment]
Firebase based notifications have only been added for sending some notifications globally. For example general information about Trakt servers problems or whatever. 

Shows channel was meant for some global notifications about shows/movies etc. but in reality it's not used at all.

**Comment by michaldrabik** [comment]
@1RandomDev I'm OK with the fork but I would like the OSS version to not have any of the premium features available then. This means also removing hidden option of unlocking after multiple clicks. 

It should just inform that premium features are only availalble in PlayStore version and thats it.

**Comment by 1RandomDev** [comment]
@michaldrabik okay I can remove the setting. I'm sure most members of the FOSS community are programmers anyway and will find a way of restoring premium functionalities after paying for the app in the play store.

And @IzzySoft how long does it usually take until an update is processed? I added another release for v3.20.1 on the 29.03. before 6pm UTC and even today it's still on the old version v3.20.0.

**Comment by michaldrabik** [comment]
@1RandomDev Naturally. Just don't want this kind of variant to be sitting out there for everybody to download.

**Comment by BradNut** [comment]
@rollingmoai  And the same can be said about many other apps that exist both as FOSS and paid Play Store. Another example being some [Simple Mobile Tools Apps](https://www.simplemobiletools.com/) where you have the [Simple Gallery Pro Play Store](https://play.google.com/store/apps/details?id=com.simplemobiletools.gallery.pro) and completely free on [GitHub](https://github.com/SimpleMobileTools/Simple-Gallery) and published to [F-Droid](https://f-droid.org/packages/com.simplemobiletools.gallery.pro/).

**Comment by McBaumwolle** [comment]
At least for users in the EU there needs to be an option to disable these analytics ([GDPR](https://gdpr-info.eu/)) and for users located in Germany the [TDDDG](https://gesetz-tdddg.de/25-tdddg/) requires informed consent - Thunderbird Mobile recently had this [issue](https://www.kuketz-blog.de/thunderbird-fuer-android-telemetrie-daten-werden-bereits-beim-start-erfasst/) (link in German) and [removed](https://github.com/thunderbird/thunderbird-android/issues/8199) them completely until further notice.

> Such access to end devices for the purposes of (web) analysis, market research and any form of advertising without informed consent is not permitted under [Section 25 (1) TDDDG](https://gesetz-tdddg.de/25-tdddg/).

This basically means that the user must be given the option to turn this on or off when the app is first started and at all times somewhere in the settings. 

For legal reasons I think @michaldrabik should look into this, Germans love their laws. :laughing:

**Comment by IzzySoft** [comment]
> This basically means that the user must be given the option to turn this on or off when the app is first started

Even stronger: they must be asked for permission **before it's turned ON** – so OFF must be the default. :man_shrugging:

**Comment by michaldrabik** [comment]
@IzzySoft @McBaumwolle Thanks, I do have it logged to implement, should do it finally.

> This basically means that the user must be given the option to turn this on or off when the app is first started

I wonder is this part Germany specific?

**Comment by michaldrabik** [comment]
Oh and this only affects PlayStore. For Github there are no trackers at all anymore.

**Comment by McBaumwolle** [comment]
The 'Telekommunikation-Digitale-Dienste-Datenschutz-Gesetz' ([TDDDG](https://de.wikipedia.org/wiki/Telekommunikation-Digitale-Dienste-Datenschutz-Gesetz?useskin=vector), we love German law names) or 'Telecommunications Telemedia Data Protection Act' (TTDSG) is a German law that specifies some things the GDPR does not - I think.

~And I thought the F-Droid version also has tracking? Or are the releases on GitHub different, aka without telemetry?~

> The OSS version available in this repo is completely free of all Google services.

Oh, never mind, I did not see that, so I'll stick to that. Nevertheless, the Play Store version should respect the TTDSG and GDPR.

**Comment by Aeris1One** [comment]
GDPR mandates the same, though it is less clearly formulated. The Thunderbird issue sent above shows it pretty clearly. So at least it is a EU-wide thing (+ some countries which have a GDPR clone like China). 

Also, I do think asking people before collecting data is way more ethical than collecting without consent, so that's a great thing to do regardless of what law applies.

**Comment by michaldrabik** [comment]
@Aeris1One Makes sense. Even if all data and crash reports in PlayStore version are fully anonymous ofc. Will be adding relevant settings in next release.

**Comment by Aeris1One** [comment]
In fact they are not that anonymous, even if there isn't the user real name in the crash report. They often contain install ID, and even if they do not, it is possible to correlate data (like phone model + language + timezone + date of install + ...) to create a unique user profile (fingerprinting).

If fingerprinting is possible, even if it's purely theorical and not done in practice, collected data are considered by GDPR as PII and all GDPR rules apply.

**Comment by McBaumwolle** [comment]
I agree - many apps simply do not care and collect anyway, so this is really nice! On the other hand, I am aware how useful crash reports are, so maybe a splash screen with something like "want to help improve the app?" when first starting (after the update) would be interesting.

**Comment by michaldrabik** [comment]
> Where is the oss version of this app. Under releases its just app-release.apk? Or am i missing something?

Yes you just download APK from latest releases.

**Comment by CodeCubeNeo** [comment]
could you maybe label the app name as app-release-oss so its more clear. If there is an automatic way?
